src/Database/Statement/StatementDecorator.php

```diff
@@ -221,11 +222,13 @@ public function fetch($type = self::FETCH_TYPE_NUM)
      * Returns the next row in a result set as an associative array. Calling this function is the same as calling
      * $statement->fetch(StatementDecorator::FETCH_TYPE_ASSOC). If no results are found false is returned.
      *
-     * @return array|false Result array containing columns and values an an associative array or false if no results
+     * @return array Result array containing columns and values an an associative array or an empty array if no results
```

Comment:
I wonder how likely it is that someone has `=== false` in their code. It might be wise to move this to 3.next for that reason.